client/resource_attribute_restriction.go

```diff
@@ -411,3 +412,30 @@ func flattenEnumChoices(choices []EnumAttributeChoice) []map[string]interface{}
 	}
 	return res
 }
+
+func expandSingleMapAllowEmpty(value interface{}) (map[string]interface{}, error) {
+	if value == nil {
+		return nil, errors.New("Value is null")
+	}
+
+	array, ok := value.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Value is not an array. Value: %v", value)
+	}
+
+	if len(array) == 0 {
+		return nil, errors.New("Array is empty")
+	}
+
+	if array[0] == nil {
+		// terraform field is defined but is empty - good enough for defining attribute types
+		return make(map[string]interface{}), nil
+	}
```

Comment:
The above check is the important part. Not sure if this is safe to add to the `expandSingleMap()` method in `structure.go` that everything else uses. I think it would be fine but haven't tested it and don't want to break everything :)

Reply:
If anyone else is interested, when a terraform field is defined like
```
resource "anaml-operations_attribute_restriction" "test_string" {
  freetext {}
}
```
then `d.Get('freetext')` will show up as `[<nil>]` in go, whereas a missing field like `d.Get('integer')` will show up as `[]`. It doesn't feel super-safe to me, but there you go...